base/shell/explorer/explorer.cpp

```diff
@@ -87,6 +87,45 @@ HideMinimizedWindows(IN BOOL bHide)
 }
 #endif
 
+static BOOL
+IsExplorerSystemShell()
+{
+    HKEY hKeyWinlogon;
+	WCHAR szShell[256];
+	WCHAR szPath[MAX_PATH];
+    LPWSTR szExplorer = NULL;
+	DWORD dwType;
+	DWORD dwBufferSize = sizeof(szShell);
+    BOOL bIsSystemShell = TRUE;
+
+    if (!GetModuleFileName(NULL, szPath, MAX_PATH))
+        return FALSE;
+
+    szExplorer = PathFindFileName((LPWSTR)szPath);
+
+	if (RegOpenKeyEx(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+		0, KEY_READ, &hKeyWinlogon) != ERROR_SUCCESS)
+    {
+        // No registry access.
+        bIsSystemShell = TRUE;
+    }
+    else
+    {
+        if (RegQueryValueEx(hKeyWinlogon, L"Shell", 0, &dwType,
+            (LPBYTE)szShell, &dwBufferSize) == ERROR_SUCCESS)
+        {
+            if (StrStrI(szShell, szExplorer))
```

Comment:
Fixing indentation (tab replaced by 4 spaces), moving the variables to where they are needed, and checking the returned type of registry value being queried:
```suggestion
    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
                      0, KEY_READ, &hKeyWinlogon) == ERROR_SUCCESS)
    {
        WCHAR szShell[256];
        DWORD dwType;
        DWORD dwBufferSize = sizeof(szShell);

        if (RegQueryValueExW(hKeyWinlogon, L"Shell", 0, &dwType,
                             (LPBYTE)szShell, &dwBufferSize) == ERROR_SUCCESS)
        {
            if ((dwType == REG_SZ || dwType == REG_EXPAND_SZ) && StrStrI(szShell, szExplorer))
```

Reply:
Can the szShell buffer be of MAX_PATH size as well?

Reply:
Maybe? I picked 256 because it's a registry key and [Microsoft](https://learn.microsoft.com/en-us/windows/win32/sysinfo/registry-element-size-limits) says the max length should be 255, although I added +1 for a terminating 0.

Reply:
Nevermind the comment I didn't reread that section of code hard enough. Going to dynamically alloc that string per tip from GeoB

Reply:
Dynamically allocated now

Reply:
Why? Just make it MAX_PATH, that is plenty.

Reply:
Win10 allows file paths longer than MAX_PATH and it is possible for one to be entered here

Reply:
In theory, sure, if the longer paths policy has been turned on. But that would break people doing GetModuleFileNameW with MAX_PATH (which this PR also does) so in practice a shell replacement can't use a path that long.

Reply:
Done in latest commit